docs/OV_Runtime_UG/auto_device_selection.md

```diff
@@ -1,332 +1,410 @@
-
-1. Check if dGPU (discrete), iGPU (integrated) and CPU devices are available
-2. Get the precision of the input model, such as FP32
-3. According to the priority of dGPU, iGPU, and CPU (in this order), if the device supports the precision of the input network, select it as the most suitable device
-
-For example, CPU, dGPU and iGPU can support the following precision and optimization capabilities:
-
-| Device | OPTIMIZATION_CAPABILITIES |
-| --- | --- |
-| CPU | WINOGRAD FP32 FP16 INT8 BIN |
-| dGPU | FP32 BIN BATCHED_BLOB FP16 INT8 |
-| iGPU | FP32 BIN BATCHED_BLOB FP16 INT8 |
-
-* When the application uses the Auto-device to run FP16 IR on a system with CPU, dGPU and iGPU, Auto-device will offload this workload to dGPU.
-* When the application uses the Auto-device to run FP16 IR on a system with CPU and iGPU, Auto-device will offload this workload to iGPU.
-* When the application uses the Auto-device to run WINOGRAD-enabled IR on a system with CPU, dGPU and iGPU, Auto-device will offload this workload to CPU.
-
-In cases when loading the network to dGPU or iGPU fails, CPU is the fall-back choice.
-
-To show the capabilities for a specific device, query the OPTIMIZATION_CAPABILITIES metric:
-
-
-```python
-from openvino.inference_engine import IECore
-
-ie = IECore()
-ie.get_metric(device_name=device,
-              metric_name="OPTIMIZATION_CAPABILITIES")
-```
-
-### Configuring the Individual Devices and Creating the Auto-Device on Top
-
-It is possible to configure each individual device as usual and create the "AUTO" device on top:
-
-```python
-from openvino.inference_engine import IECore
-
-ie = IECore()
-net = ie.read_network(model=path_to_model)
-
-cpu_config = {}
-gpu_config = {}
-
-ie.set_config(config=cpu_config, device_name="CPU")
-ie.set_config(config=gpu_config, device_name="GPU")
-
-# Load the network to the AUTO device
-exec_net = ie.load_network(network=net, device_name="AUTO")
-```
-
-Alternatively, you can combine all the individual device settings into single config file and load it, allowing the Auto-device plugin to parse and apply it to the right devices. See the code example here:
+For C++ API
+@sphinxdirective
+.. code-block:: sh
 
-```python
-from openvino.inference_engine import IECore
+   ov::runtime::Core::get_available_devices() (see Hello Query Device C++ Sample)
+@endsphinxdirective
 
-# Init the Inference Engine Core
-ie = IECore()
+For Python API
+@sphinxdirective
+.. code-block:: sh
 
-# Read a network in IR or ONNX format
-net = ie.read_network(model=path_to_model)
+   openvino.runtime.Core.available_devices (see Hello Query Device Python Sample)
+@endsphinxdirective
 
-full_config = {}
 
-# Load the network to the AUTO device
-exec_net = ie.load_network(network=net, device_name="AUTO", config=full_config)
-```
+### Performance Hints
+The `ov::hint` property enables you to specify a performance mode for the plugin to be more efficient for particular use cases.
```

Comment:
Why is it described in "auto device selection" doc only? Is performance hint supported for AUTO plugin only? @andrew-zaytsev

Reply:
Developer need to use performance hint instead of detail configurations for AUTO plugin. So we high light hints included refence code in AUTO documents.

Performance hints are also support in CPU plugin or GPU plugin in 2022.1 release.

Reply:
`ov::hint` is NOT a property. Use `ov::hint::performance_mode`